cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -1142,7 +1127,7 @@ Slice::CsGenerator::writeSequenceMarshalUnmarshalCode(
     assert(cont);
     if (useHelper)
     {
-        string helperName = getUnqualified(getNamespace(seq) + "." + seq->name() + "Helper", scope);
+        string helperName = getUnqualified(seq, scope, "", "Helper");
```

Comment:
We already have an overload of `getUnqualified` which handles applying suffixes to types.